sim/include/component/pos2DComponent.h

```diff
@@ -1,12 +1,36 @@
 #ifndef BENTOBOX_POS2DCOMPONENT_H
 #define BENTOBOX_POS2DCOMPONENT_H
 
+#include <bento/protos/ecs.pb.h>
+#include <component/userComponent.h>
+
 namespace ics::component {
-struct Pos2DComponent : public ics::BaseComponent {
-    unsigned int x;
-    unsigned int y;
+const char POS2D_COMPONENT_NAME[] = "Pos2DComponent";
+}
+
+namespace {
+
+// This needs to be in the header as it is called immediately in the header as
+// well.
+bento::protos::ComponentDef createCompDef() {
+    auto compDef = bento::protos::ComponentDef();
+    compDef.set_name(ics::component::POS2D_COMPONENT_NAME);
+
+    auto& schema = *compDef.mutable_schema();
+    schema["x"] = bento::protos::Type();
+    schema["x"].set_primitive(bento::protos::Type_Primitive_INT64);
+
+    schema["y"] = bento::protos::Type();
+    schema["y"].set_primitive(bento::protos::Type_Primitive_INT64);
 
-    Pos2DComponent(unsigned int x, unsigned int y) : x(x), y(y) {}
+    return compDef;
+};
+
+}  // namespace
+
+namespace ics::component {
+struct Pos2DComponent : public UserComponent {
+    Pos2DComponent() : UserComponent(POS2D_COMPONENT_NAME, createCompDef()) {}
```

Comment:
What happens when you import both `pos1DComponent.h` and `pos2DComponent.h` at the same time? Does `createCompDef()` do the right thing?

Reply:
`createCompDef` of both components are defined in a `namespace {}` scope, this means that it doesn't export function.